test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1

```diff
@@ -5,219 +5,488 @@
+                ExpectedFullyQualifiedErrorId = "FormatHexOnlySupportsFileSystemPaths,Microsoft.PowerShell.Commands.FormatHex"
+            }
+            @{
+                Name = "If given a non FileSystem path in array (with LiteralPath), continues to process valid paths 'fhx -Path `$invalidPath, `$inputFile1  -ev e -ErrorAction SilentlyContinue'"
+                InvalidPath = "Cert:\CurrentUser\My\$thumbprint"
+                ExpectedFullyQualifiedErrorId = "FormatHexOnlySupportsFileSystemPaths,Microsoft.PowerShell.Commands.FormatHex"
+            }
+        )
+
+        foreach ($testCase in $testCases)
+        {
+            RunContinuesToProcessCase $testCase
         }
     }
 
-    # This test is to validate the alias for Format-Hex cmdlet.
-    It "ValidateCmdletAlias" {
+    Context "Cmdlet Functionality" {
+
+        It "Path is default Parameter Set 'fhx `$inputFile1'" {
+
+            $result =  Format-Hex $inputFile1
 
-        try
-        {
-            $result = Get-Command fhx -ErrorAction Stop
             $result | Should Not BeNullOrEmpty
-            $result.CommandType.ToString() | Should Be "Alias"
+            ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
+            $actualResult = $result.ToString()
+            $actualResult | Should Match $inputText1
         }
-        catch
-        {
-            $_ | Should BeNullOrEmpty
+
+        It "Validate file input from Pipeline 'Get-ChildItem `$inputFile1 | Format-Hex'" {
+
+            $result = Get-ChildItem $inputFile1 | Format-Hex
+
+            $result | Should Not BeNullOrEmpty
+            ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
+            $actualResult = $result.ToString()
+            $actualResult | Should Match $inputText1
+        }
+
+        It "Validate that streamed text does not have buffer underrun problems ''a' * 30 | Format-Hex'" {
+
+            $result = "a" * 30 | Format-Hex
+
+            $result | Should Not BeNullOrEmpty
+            ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
+            $result.ToString() | Should be "00000000   61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61  aaaaaaaaaaaaaaaa$($newLine)00000010   61 61 61 61 61 61 61 61 61 61 61 61 61 61        aaaaaaaaaaaaaa  "
+        }
+
+        It "Validate that files do not have buffer underrun problems 'Format-Hex -path `$InputFile2'" {
+
+            $result = Format-Hex -path $InputFile2
+
+            $result | Should Not BeNullOrEmpty
+            $result.Count | should be 2
+            $result[0].ToString() | Should be "00000000   54 68 69 73 20 69 73 20 61 20 62 69 74 20 6D 6F  This is a bit mo"
+            $result[1].ToString() | Should be "00000010   72 65 20 74 65 78 74                             re text         "
+        }
+
+        It "Validate the alias (fhx) for Format-Hex cmdlet is available" {
+
+            { Get-Alias fhx } | Should Not Throw
```

Comment:
Please remove the test because we already have separate files with tests for aliases.